recon_surf/recon-surf.sh

```diff
@@ -540,18 +518,31 @@ echo " " |& tee -a $LF
 echo "================== Creating orig and rawavg from input =========================" |& tee -a $LF
 echo " " |& tee -a $LF
 
+CONFORM_LF=$SUBJECTS_DIR/$subject/scripts/conform.log
+if [ $CONFORM_LF != /dev/null ] ; then  rm -f $CONFORM_LF ; fi
+echo "Log file for Conform test" > $CONFORM_LF
+
 # check for input conformance
-cmin=""
-if  [ "$vox_size" == "auto" ]
+cmd="$python ${binpath}../FastSurferCNN/data_loader/conform.py -i $t1 --check_only --vox_size min --verbose"
+RunIt "$cmd" "$LF -a $CONFORM_LF"
+
+# look into the CONFORM_LF to find the voxel sizes, the second conform.py call will check the legality of vox_size
+vox_size=`cat $CONFORM_LF | grep -E " - Voxel Size " | cut -d' ' -f5 | cut -d'x' -f1`
+# remove the temporary conform_log (all info is also in the recon-surf logfile)
+if [ -f "$CONFORM_LF" ]; then rm -f $CONFORM_LF ; fi
+
+# here, we check the correct vox_size by passing it to the next conform, so errors in this line might be caused above
+cmd="$python ${binpath}../FastSurferCNN/data_loader/conform.py -i $aparc_aseg_segfile --check_only --vox_size $vox_size --dtype any --verbose"
+RunIt "$cmd" $LF
+
+if [ "$vox_size" -lt "$hires_voxsize_threshold" ]
```

Comment:
Why is this if block necessary? Does not really make sense to me

Reply:
or at least also set the flags for conform min and hires here, so we can avoid the other if [ "$vox_size" -lt "$hires_voxsize_threshold" ]

Reply:
(i.e. instead of 
line 637:  cm="-noconform" else cm=""
line 727: hsuff=".predec" else hsuff="")

Reply:
While in the following code, we could also test against `$hiresflag`, here we need to SET the hiresflag variable.
Also, I would want to have a string in the log somewhere that documents which mode was selected (although that would also always be apparent by the existence of `-hires` in later calls. Still having this here and also the `$hires_voxsize_threshold` documented seems like a good idea to me.

I'll push a patch to clean up the ifs.

Reply:
The voxsize_threshold is set to 0.999. This can not be changed. So in my opinion this just complicates recon-surf unnecessarily. Also, either the input is 1 or below 1 = hires processing. Having variable voxelsizes already does not make sense here. I would just have a hires flag and turn this on if the voxsize flag in run_fastsurfer.sh is set accordingly. Then have a check in conform.py --check_only if the hires flag is passed and resolution is below 1.0 mm and abort if this is not the case. This only makes sense if you have a variable hires_voxsize_threshold to run 1.0 mm settings with hires images, but this is not tested (neither for FastSurfer nor FreeSurfer). So do we really want to support this?